lib/ts/recipe/emailpassword/api/implementation.ts

```diff
@@ -737,18 +825,46 @@ export default function getAPIImplementation(): APIInterface {
                 throw new Error("Race condition error - please call this API again");
             }
 
-            let session = await Session.createNewSession(
-                options.req,
-                options.res,
+            if (mfaInstance === undefined) {
+                // No MFA stuff here, so we just create and return the session
+                let session = await Session.createNewOrKeepExistingSession(
+                    options.req,
+                    options.res,
+                    tenantId,
+                    emailPasswordRecipeUser.recipeUserId,
+                    {},
+                    {},
+                    userContext
+                );
+
+                return {
+                    status: "OK",
+                    session,
+                    user: (await getUser(response.user.id, userContext))!, // fetching user again cause the user might have been updated while setting up mfa
```

Comment:
the comment here is wrong.. it's in the branch for mfaInstance === undefined, but it says "updated while setting up mfa"

Reply:
Done